### PR TITLE
[webgpu] Enable bool and add op logicalAnd

### PR DESCRIPTION
**Pull request by axinging**
FEATURE
From: https://github.com/tensorflow/tfjs/issues/5180

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5190)

### CODE REVIEW

**Comment by axinging** [comment]
@qjia7 @haoyunfeix @xhcao @gyagp, PTAL

File: tfjs-backend-webgpu/src/backend_webgpu.ts [hunk 1]
```diff
@@ -242,6 +242,11 @@ export class WebGPUBackend extends KernelBackend {
     const byteSize =
         util.sizeFromShape(shape) * webgpu_util.GPUBytesPerElement(dtype);
 
+    // bool is stored in Uint8Array, converted it to Int32Array.
+    if (values instanceof Uint8Array) {
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/backend_webgpu.ts:246]: `if (dtype === 'bool' && values instanceof Uint8Array)`?

File: tfjs-backend-webgpu/src/kernels/binary_ops.ts [hunk 2]
```diff
@@ -88,6 +89,11 @@ export function getBinaryOpString(
     case BinaryOpType.LESS_EQUAL:
       return useVec4 ? 'return vec4(lessThanEqual(a, b));' :
                        'return float(a <= b);';
+    case BinaryOpType.LOGICAL_AND:
+      return useVec4 ? `return vec4(
+      vec4(greaterThanEqual(a, vec4(1.0))) *
+      vec4(greaterThanEqual(b, vec4(1.0))));` :
+                       'return float(float(a) >= 1.0 && float(b) >= 1.0);';
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/binary_ops.ts:96]: Can directly use `return float(a >= 1.0 && b >= 1.0);`? `float(a)` and `float(b)` are not needed?

File: tfjs-backend-webgpu/src/setup_test.ts [hunk 3]
```diff
@@ -403,40 +399,35 @@ const TEST_FILTERS: TestFilter[] = [
...
-      '6D',                        // Rank 5 is not yet implemented.
-      'accepts tensor with bool',  // Actual != Expected.
-      'gradient',                  // zerosLike not yet implemented.
+      '5D',        // Rank 5 is not yet implemented.
+      '6D',        // Rank 5 is not yet implemented.
+      'gradient',  // zerosLike not yet implemented.
     ]
   },
   {
     include: 'abs',
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/setup_test.ts:424]: nit: Consider to use `startsWith: 'abs ',` so that the `excludes` can be further simplified.

**Review by qjia7 — APPROVED** [review]
LGTM with nits

**Comment by qjia7** [comment]
@lina128 @jinjingforever please take a look, thanks.

File: tfjs-backend-webgpu/src/kernels/binary_ops.ts (hunk 2 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/binary_ops.ts:96]: Remove float results in below fail:
FAILED logicalAnd test-webgpu {"WEBGPU_CPU_FORWARD":false} Tensor1D.
FAILED logicalAnd test-webgpu {"WEBGPU_CPU_FORWARD":false} Tensor2D
FAILED logicalAnd test-webgpu {"WEBGPU_CPU_FORWARD":false} broadcasting Tensor2D shapes
FAILED logicalAnd test-webgpu {"WEBGPU_CPU_FORWARD":false} Tensor3D
FAILED logicalAnd test-webgpu {"WEBGPU_CPU_FORWARD":false} broadcasting Tensor3D shapes
FAILED logicalAnd test-webgpu {"WEBGPU_CPU_FORWARD":false} broadcasting Tensor4D shapes
FAILED logicalAnd test-webgpu {"WEBGPU_CPU_FORWARD":false} TensorLike
FAILED logicalAnd test-webgpu {"WEBGPU_CPU_FORWARD":false} TensorLike Chained

File: tfjs-backend-webgpu/src/backend_webgpu.ts (hunk 1 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/backend_webgpu.ts]: done

File: tfjs-backend-webgpu/src/setup_test.ts (hunk 3 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/setup_test.ts]: done

File: tfjs-backend-webgpu/src/kernels/binary_ops.ts (hunk 2 above)
Comment (qjia7) [reply tfjs-backend-webgpu/src/kernels/binary_ops.ts:96]: Ack. But it's weird. Maybe the underlying shader is not correctly translated.

**Review by qjia7 — APPROVED** [review]
Still LGTM, thanks.

**Review by jinjingforever — APPROVED** [review]

**Pull request merged**